src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -94,16 +131,22 @@ private function completeDefinition($id, Definition $definition)
 
                 if (isset($this->types[$typeHint->name])) {
                     $value = new Reference($this->types[$typeHint->name]);
+                    $addMethodCall = true;
                 } else {
                     try {
                         $value = $this->createAutowiredDefinition($typeHint, $id);
+                        $addMethodCall = true;
                     } catch (RuntimeException $e) {
                         if ($parameter->allowsNull()) {
                             $value = null;
                         } elseif ($parameter->isDefaultValueAvailable()) {
                             $value = $parameter->getDefaultValue();
                         } else {
-                            throw $e;
+                            if (1 === $e->getCode() || $isConstructor) {
```

Comment:
I was thinking about introducing a `const` in `RuntimeException` for this value but it's maybe too intrusive.